lib/src/fortio.cpp

```diff
@@ -409,3 +409,67 @@ int eclfio_put( std::FILE* fp,
 } catch( std::exception& ) {
     return ECL_ERR_SEEK;
 }
+
+int ecl_default_blocksize( int type ) {
+    switch( type ) {
+        case ECL_BLOCKSIZE_NUMERIC: return ECL_DEFAULT_BLOCKSIZE_NUMERIC;
+        case ECL_BLOCKSIZE_STRING:  return ECL_DEFAULT_BLOCKSIZE_STRING;
+        default:                    return -1;
+    }
+}
+
+int eclfio_array_get( std::FILE* fp,
+                      const char* opts,
+                      int len,
+                      int nmemb,
+                      void* array ) {
+
+    if( nmemb < 0 ) return ECL_EINVAL;
+
+    if( len < 1 ) len = 1;
```

Comment:
What is the logic here?

Reply:
Just robustness, allowing zero or negative values as "don't worry about len at all), to avoid nasty divide-by-zeros.